lib/thor/shell/printer.rb

```diff
@@ -0,0 +1,13 @@
+class Thor
+  module Shell
+    class Printer
```

Comment:
Do we need this class? Can't we just merge it with ColumnPrinter?

Reply:
`TablePrinter` and `WrappedPrinter` inherit from `Printer` as well.
All three subclasses implement the `print` method in their own way.

Reply:
Can't we change them to inherit from `ColumnPrinter`?

Reply:
I'm asking this because Printer doesn't have the same interface as their subclasses so it is breaking the Liskov substitution principle, and I see no reason to break that.

Reply:
Done.